client/my-sites/controller.js

```diff
@@ -137,7 +157,7 @@ function onSelectedSiteAvailable( context ) {
 
 	if ( isDomainOnlySite( state, selectedSite.ID ) &&
 		! isPathAllowedForDomainOnlySite( context.pathname, selectedSite.slug ) ) {
```

Comment:
`isPathAllowedForDomainOnlySite` allows:
domains/manage/test.wordpress.com/email/test.wordpress.com
domains/manage/test.wordpress.com/email-forwarding/test.wordpress.com

Would be nice to check if we allow those, and to actually go through all of the paths, as we have transfer between blogs, and transfer between users that start with `/domains/manage`.